<?xml version="1.0"?>
<launch>
  <arg name="input/pointcloud" default="/sensing/lidar/top/pointcloud"/>
  <arg name="output/pointcloud/segmentation" default="segmentation"/>
  <arg name="output/pointcloud/visualization" default="visualization"/>
  <arg name="output/pointcloud/filtered" default="filtered"/>
  <arg name="data_path" default="$(env HOME)/autoware_data" description="packages data and artifacts directory path"/>
  <arg name="model_name" default="frnet" description="options: `frnet`"/>
  <arg name="model_path" default="$(var data_path)/lidar_frnet"/>
  <arg name="model_param_path" default="$(find-pkg-share autoware_lidar_frnet)/config/$(var model_name).param.yaml"/>
  <arg name="ml_package_param_path" default="$(var model_path)/ml_package_$(var model_name).param.yaml"/>
  <arg name="diagnostics_param_path" default="$(find-pkg-share autoware_lidar_frnet)/config/diagnostics_frnet.param.yaml"/>
  <arg name="build_only" default="false" description="shutdown node after TensorRT engine file is built"/>
  <arg name="use_pointcloud_container" default="false" description="use pointcloud_container"/>
  <arg name="pointcloud_container_name" default="pointcloud_container" description="pointcloud_container name"/>

  <group if="$(var use_pointcloud_container)">
    <load_composable_node target="$(var pointcloud_container_name)">
      <composable_node pkg="autoware_lidar_frnet" plugin="autoware::lidar_frnet::LidarFRNetNode" name="lidar_frnet">
        <remap from="~/input/pointcloud" to="$(var input/pointcloud)"/>
        <remap from="~/output/pointcloud/segmentation" to="$(var output/pointcloud/segmentation)"/>
        <remap from="~/output/pointcloud/visualization" to="$(var output/pointcloud/visualization)"/>
        <remap from="~/output/pointcloud/filtered" to="$(var output/pointcloud/filtered)"/>
        <param from="$(var model_param_path)" allow_substs="true"/>
        <param from="$(var ml_package_param_path)"/>
        <param from="$(var diagnostics_param_path)"/>
        <param name="build_only" value="$(var build_only)"/>
      </composable_node>
    </load_composable_node>
  </group>
  <group unless="$(var use_pointcloud_container)">
    <node pkg="autoware_lidar_frnet" exec="autoware_lidar_frnet_node" name="lidar_frnet" output="screen">
      <remap from="~/input/pointcloud" to="$(var input/pointcloud)"/>
      <remap from="~/output/pointcloud/segmentation" to="$(var output/pointcloud/segmentation)"/>
      <remap from="~/output/pointcloud/visualization" to="$(var output/pointcloud/visualization)"/>
      <remap from="~/output/pointcloud/filtered" to="$(var output/pointcloud/filtered)"/>
      <param from="$(var model_param_path)" allow_substs="true"/>
      <param from="$(var ml_package_param_path)"/>
      <param from="$(var diagnostics_param_path)"/>
      <param name="build_only" value="$(var build_only)"/>
    </node>
  </group>
</launch>
